test/sequential/test-watch-mode.mjs

```diff
@@ -174,6 +175,41 @@ describe('watch mode', { concurrency: true, timeout: 60_0000 }, () => {
     });
   });
 
+  it('should watch changes to previously loaded dependencies', async () => {
+    const dependencyContent = 'module.exports = {}';
+    const dependency = createTmpFile(dependencyContent);
+    const relativeDependencyPath = `./${path.basename(dependency)}`;
+    const dependant = createTmpFile(`console.log(require('${relativeDependencyPath}'))`);
+
+    let stderr = '';
+    let stdout = '';
+    const child = spawn(execPath, ['--watch', '--no-warnings', dependant], { encoding: 'utf8' });
+    child.stdout.on('data', (data) => { stdout += data; });
+    child.stderr.on('data', (data) => { stderr += data; });
+    child.on('error', (err) => { throw err; });
+
+    await once(child.stdout, 'data');
+    rmSync(dependency, { force: true });
+
+    await setTimeout(600); // throttle + 100
+    writeFileSync(dependency, dependencyContent);
+
+    await setTimeout(600); // throttle + 100
+    child.kill();
+    await once(child, 'exit');
+
+    // TODO(ruyadorno): fs.watch is flaky when removing files from a watched
+    // folder, in this test we want to assert the expected reload happened
+    // after a missing file is recreated so we'll skip the assertions in case
+    // the expected reload+failure never happened.
+    // This should be an assertion for the failure instead of an if block once
+    // the source of this flakyness gets resolved.
+    if (stdout.match(/Failed/g)?.length) {
+      assert.strictEqual(stdout.split('Failed')[1].match(/Restarting/g)?.length, 1, new Error('should have restarted after fail'));
+      assert.ok(stderr.match(/MODULE_NOT_FOUND/g)?.length, new Error('should have failed for not finding the removed file'));
```

Comment:
```suggestion
    if (stdout.includes('Failed')) {
      assert.ok(stdout.split('Failed')[1].includes('Restarting'), new Error('should have restarted after fail'));
      assert.ok(stderr.includes('MODULE_NOT_FOUND'), new Error('should have failed for not finding the removed file'));
```
We should use `%String.prototype.includes%` here instead because we're only checking if those values are included, although for the first **assert** we can keep it as is if we really need to check if `Restarting` only appears once.

Reply:
Or `assert.match`, to get a helpful string comparison between what was expected in case the test fails.